wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/Views/DeviceInfoViewController.swift

```diff
@@ -0,0 +1,97 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Combine
+import WireCommonComponents
+import WireDataModel
+import SwiftUI
+
+/// A customized hosting controller for `DeviceDetailsView` and `ProfileDeviceDetailsView` in order to allow displaying
+/// a custom navigation item title view and a debug menu button in the navigation bar.
+final class DeviceInfoViewController<Content>: UIHostingController<Content> where Content: DeviceInfoView {
+
+    private var cancellables = Set<AnyCancellable>()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupNavigationItemTitleObservation()
+        setupDebugMenuButtonIfNeeded()
+    }
+
+    private func setupNavigationItemTitleObservation() {
+
+        let certificatePublisher = rootView.viewModel.$e2eIdentityCertificate
+        let isProtuesVerifiedPublisher = rootView.viewModel.$isProteusVerificationEnabled
+        certificatePublisher.combineLatest(isProtuesVerifiedPublisher)
+            .sink { [weak self] certificate, isProteusVerified in
+                self?.updateNavigationItemTitle(certificate, isProteusVerified)
+            }
+            .store(in: &cancellables)
+    }
+
+    private func updateNavigationItemTitle(
+        _ certificate: E2eIdentityCertificate?,
+        _ isProteusVerified: Bool
+    ) {
+
+        let deviceName = NSMutableAttributedString(string: rootView.viewModel.title)
```

Comment:
In my opinion we could also have a property to the `viewModel` in this view controller, since it's also part of the view and simplifies the access a bit.